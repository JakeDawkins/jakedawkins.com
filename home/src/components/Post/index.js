import React, { PropTypes } from "react";
import ReactMarkdown from "react-markdown";
import { Link } from "react-router";
import "../../styles/_post.scss";

import Author from "../Author";
import data from "../../api/data.js"; //data for author

const Post = (props) =>
  <section className="post" itemScope itemType="http://schema.org/BlogPosting">
      <article role="article" id="scroll" className="post-content" itemProp="articleBody">
        { props.disableBackLink ? null :
          <p>
            <Link to={props.backLinkUrl ? props.backLinkUrl : "/posts"}>&lt;&nbsp;&nbsp;Back to Posts</Link>
          </p>
        }
        <ReactMarkdown source={props.markdown} />
        { props.disableBackLink ? null :
          <p style={{textAlign: "center"}}>
            <Link to="/posts">&lt;&nbsp;&nbsp;Back to Posts</Link>
          </p>
        }
      </article>

      {
        props.disableAuthor
          ? null
          : <Author
            userDisplayName={data.user.name}
            userImageUrl={data.user.image}
            userDescription={data.user.user_description}
            userEmail={data.user.email}
            />
      }
  </section>

//backLinkUrl not used yet
Post.propTypes = {
  markdown: PropTypes.string.isRequired,
  backLinkUrl: PropTypes.string,
  disableBackLink: PropTypes.bool,
  disableAuthor: PropTypes.bool,
}

export default Post;
