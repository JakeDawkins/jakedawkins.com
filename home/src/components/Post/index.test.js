import Post from './';
import React from 'react';
import renderer from 'react-test-renderer';

describe("Post", () => {

  it("should exist", () => {
    const post = renderer.create(
      <Post
        markdown="hello\n\nworld"
      />
    );

    expect(post).toMatchSnapshot();
  });

  it("should render props correctly", () => {
    const post = renderer.create(
      <Post
        markdown="hello\n\nworld"
      />
    );

    const postWithProps = renderer.create(
      <Post
        markdown="hello\n\nworld"
        backLinkUrl="/testback"
        disableBackLink={true}
        disableAuthor={true}
      />
    );

    expect(postWithProps).toMatchSnapshot();
    expect(postWithProps).not.toBe(post);
  });

});
