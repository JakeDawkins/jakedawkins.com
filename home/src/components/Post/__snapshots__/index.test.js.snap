exports[`Post should exist 1`] = `
<section
  className="post"
  itemScope={true}
  itemType="http://schema.org/BlogPosting">
  <article
    className="post-content"
    id="scroll"
    itemProp="articleBody"
    role="article">
    <p>
      <a
        onClick={[Function bound handleClick]}
        style={Object {}}>
        <  Back to Posts
      </a>
    </p>
    <div>
      <p>
        hello\\n\\nworld
      </p>
    </div>
    <p
      style={
        Object {
          "textAlign": "center"
        }
      }>
      <a
        onClick={[Function bound handleClick]}
        style={Object {}}>
        <  Back to Posts
      </a>
    </p>
  </article>
  <section
    className="author"
    itemProp="author">
    <div
      className="details"
      itemScope={true}
      itemType="http://schema.org/Person">
      <img
        alt="Jake Dawkins"
        className="img-rounded"
        itemProp="image"
        src="http://content.jakedawkins.com/layout/selfportrait.png" />
      <p
        className="def">
        Author
      </p>
      <h3
        className="name">
        <a
          itemProp="name"
          onClick={[Function bound handleClick]}
          style={Object {}}>
          Jake Dawkins
        </a>
      </h3>
      <p
        className="desc">
        Developer | Engineer
      </p>
      <a
        className="email"
        href="mailto:dawkinsjh@gmail.com"
        itemProp="email">
        dawkinsjh@gmail.com
      </a>
    </div>
  </section>
</section>
`;

exports[`Post should render props correctly 1`] = `
<section
  className="post"
  itemScope={true}
  itemType="http://schema.org/BlogPosting">
  <article
    className="post-content"
    id="scroll"
    itemProp="articleBody"
    role="article">
    <div>
      <p>
        hello\\n\\nworld
      </p>
    </div>
  </article>
</section>
`;
